scrape/scrape.go

```diff
@@ -714,13 +715,19 @@ type scrapeCache struct {
 	// seriesCur and seriesPrev store the labels of series that were seen
 	// in the current and previous scrape.
 	// We hold two maps and swap them out to save allocations.
-	seriesCur  map[uint64]labels.Labels
-	seriesPrev map[uint64]labels.Labels
+	seriesCur  map[uint64]seriesEntry
```

Comment:
Putting this in here feels weird, why is staleness being mixed with metadata?

Why is the cache entry no enough?

Reply:
So the main reason this was required is that we write a "stale" datapoint for a series using appender "Add(...)" which requires the full labels, metadata, etc to need on hand when we call that appender method from within "forEachStale(...)".

Reply:
Do you think we should revert to zero-ing the metadata if a scrape is stale? Or calling "Add(...)" with previously observed metadata is correct here? (in the calls to ".Add(...)" from within ".forEachStatel(...)" callback)

Reply:
The metadata is still correct if a scrape fails, only if there is new (or missing) metadata should Add need to get called again.

Reply:
The use case for this is we don't have a "Ref" kept around. Also even if we have a "Ref" and can use "AddFast" what happens if "ErrNotFound" is returned and we have to call "Add(...)" from this code which already is calling just "Add(...)" directly?
https://github.com/prometheus/prometheus/blob/2b75c1b19995377f2ffcea7aff6818982f2f2bc6/scrape/scrape.go#L1345-L1355

`seriesCur` and `seriesPrev` back what is passed into that inner method:
https://github.com/prometheus/prometheus/blob/2b75c1b19995377f2ffcea7aff6818982f2f2bc6/scrape/scrape.go#L832-L840